### ISSUE TITLE
kernel 5.15.120 crash with cilium 1.13.4

### ISSUE THREAD

**Issue by livelace** [cilium/cilium#26756]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Recently I decided to upgrade cilium from 1.11.8 to 1.13.4.
Right after kernel 5.15.120 started to crash:

```
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.179134] ------------[ cut here ]------------
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.179683] WARNING: CPU: 2 PID: 18279 at kernel/trace/bpf_trace.c:684 bpf_event_output+0x1c9/0x1f0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.189785] CPU: 2 PID: 18279 Comm: helm Not tainted 5.15.120-gentoo #1
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.190621] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS 1.16.0-20220807_005459-localhost 04/01/2014
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.192697] Code: ff 65 ff 0d 29 9d f8 55 65 48 8b 0c 25 28 00 00 00 48 3b 4d d0 75 30 48 83 c4 50 5b 41 5c 41 5d 41 5e 41 5f 5d c3 cc cc cc cc <0f> 0b 48 c7 c0 f0 ff ff ff eb cc 48 c7 c0 f9 ff ff ff eb c3 48 c7
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.197786] RBP: ffffbd398172fc80 R08: 0000000000000000 R09: 0000000000000000
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.199398] R13: ffffbd398172fd58 R14: 00000000000001bb R15: 0000000000000000
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.200117] FS:  000000c000335090(0000) GS:ffff92e597a80000(0000) knlGS:0000000000000000
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.202629] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.204635]  <TASK>
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.204864]  ? __warn+0xc3/0x190
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.205202]  ? bpf_event_output+0x1c9/0x1f0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.206298]  ? handle_bug+0x42/0x70
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.206655]  ? exc_invalid_op+0x16/0x50
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.207021]  ? asm_exc_invalid_op+0x16/0x20
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.207462]  ? 0xffffffffc0aee814
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.207784]  ? bpf_event_output+0x1c9/0x1f0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.208179]  ? enqueue_task_fair+0x3da/0x7b0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.208643]  bpf_event_output_data+0x27/0x40
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.209859]  ? migrate_disable+0x5b/0x80
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.210445]  __cgroup_bpf_run_filter_sock_addr+0x198/0x260
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.211245]  ? __local_bh_enable_ip+0x83/0xc0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.211973]  ? trace_hardirqs_on+0x33/0x50
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.212490]  ? __local_bh_enable_ip+0x83/0xc0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.212995]  ? lock_sock_nested+0xbf/0xf0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.213435]  ? sock_getsockopt+0x93a/0xaf0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.213847]  inet_getname+0xf7/0x100
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.214235]  __sys_getpeername+0x8c/0xf0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.214674]  ? __fget_files+0xa4/0xc0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.215088]  ? check_preemption_disabled+0x35/0xe0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.215726]  ? syscall_enter_from_user_mode+0x1a/0x1c0
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.216354]  ? trace_hardirqs_on+0x33/0x50
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.216975]  __x64_sys_getpeername+0x17/0x20
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.217560]  do_syscall_64+0x41/0x90
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.217978]  entry_SYSCALL_64_after_hwframe+0x61/0xcb
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.218611] RIP: 0033:0x4b0bf6
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.218916] Code: cc cc cc cc cc cc cc cc cc cc cc cc cc cc cc cc cc cc cc cc 48 8b 7c 24 10 48 8b 74 24 18 48 8b 54 24 20 48 8b 44 24 08 0f 05 <48> 3d 01 f0 ff ff 76 1b 48 c7 44 24 28 ff ff ff ff 48 c7 44 24 30
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.220931] RSP: 002b:000000c0004747a0 EFLAGS: 00000202 ORIG_RAX: 0000000000000034
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.222754] RDX: 000000c00047481c RSI: 000000c000474820 RDI: 0000000000000006
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.225401] R13: 0000000000000000 R14: 000000c0000c6d00 R15: ffffffffffffffff
Jul  9 21:33:27 k8s-1-master1 kernel: [  648.226160]  </TASK>
```

Crashing happens not immediately, but after  ~12-16 hours. I moved back to 1.11.8 and crashes dissappeared.

### Cilium Version

1.13.4

### Kernel Version

5.15.120

### Kubernetes Version

1.20.4

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by julianwiedmann** [comment]
That's https://elixir.bootlin.com/linux/v5.15.120/source/kernel/trace/bpf_trace.c#L684

```
	if (WARN_ON_ONCE(nest_level > ARRAY_SIZE(bpf_misc_sds.sds))) {
		ret = -EBUSY;
		goto out;
	}
```

@borkmann ring any bells?

**Comment by borkmann** [comment]
> Right after kernel 5.15.120 started to crash:

@livelace was there a stable kernel you ran previously where this did not trigger?

**Comment by borkmann** [comment]
triggered by: https://lore.kernel.org/bpf/20190925234312.94063-2-allanzhang@google.com/

**Comment by olsajiri** [comment]
> Crashing happens not immediately, but after ~12-16 hours. I moved back to 1.11.8 and crashes dissappeared.

@livelace any chance there's some stacktrace or anything on the console when the crash itself happened?

thanks,
jirka

**Comment by livelace** [comment]
@borkmann 5.15.120 is quite stable, I didn't have any problems with previous kernel versions too.
Crash appeared only after upgrade cilium version.

**Comment by livelace** [comment]
@olsajiri no chances. i switched back to previous version because it's a production environment.

**Comment by livelace** [comment]
current uptime is 4 days, cilium version 1.11.8 (old).

**Comment by borkmann** [comment]
@livelace it's certainly a kernel issue, just that the Cilium code seems to uncover it. After that warn happened, did the Hubble / monitor output stopped completely showing no traffic, or did it still show traffic?

**Comment by livelace** [comment]
@borkmann kernel just gone to panic and machine rebooted.

**Comment by olsajiri** [comment]
@livelace do you have CONFIG_PREEMPT_RCU enabled in your kernel config?

**Comment by livelace** [comment]
@olsajiri Yep

**Comment by DonOtuseGH** [comment]
Hello,

we performed an update of some 3 node kubernetes clusters from:

```
openSUSE   15.4
kernel     5.14.21-150400.24.66-default
kubernetes v1.25.11
containerd v1.6.21
cilium     1.12.11
```

to:


```
openSUSE   15.5
kernel     5.14.21-150500.55.19-default
kubernetes v1.27.5 (via v1.26.6, v1.27.3)
containerd v1.6.21
cilium     1.14.1 (via 1.13.4, 1.13.6)
```

After the upgrade we see node reboots due to a kernel panic, sometimes once or twice a day. I found this issue and i wonder if this could fit to our problem as well?

We have nothing in the nodes journal/syslog/kernlog or cilium-agent/-operator log, which points to the panic and the kernel doesn't print a stack trace either.

Would it help to create a new issue for further investigation, comment here? Or will there be a new release available soon including the fix, which we can install and verify?

Thanks and best regards, Don.

**Comment by julianwiedmann** [comment]
> Would it help to create a new issue for further investigation, comment here? Or will there be a new release available soon including the fix, which we can install and verify?

The issue reported here is resolved by a kernel fix: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=f2c67a3e60d1071b65848efaa8c3b66c363dd025

So you would need to pick up a kernel update with this patch.

**Comment by julianwiedmann** [comment]
(closing this issue, just a matter of distros picking up the kernel fix)
